src/Symfony/Component/Validator/Validator/RecursiveContextualValidator.php

```diff
@@ -277,11 +277,15 @@ public function getViolations()
      */
     protected function normalizeGroups($groups)
     {
-        if (\is_array($groups)) {
-            return $groups;
+        if (!\is_array($groups)) {
+            $groups = [$groups];
         }
 
-        return [$groups];
+        if (\in_array(null, $groups, true)) {
+            throw new ValidatorException('Value in array of validation groups cannot be null.');
```

Comment:
Should be an InvalidArgumentException I suppose?
What about other types that are neither strings nor GroupSequence?
Also, since this is an improvement, it should target 6.3.